tests/puppeteer-attachment.spec.ts

```diff
@@ -0,0 +1,178 @@
+#!/usr/bin/env -S node --no-warnings --loader ts-node/esm
+/**
+ *   Wechaty - https://github.com/chatie/wechaty
+ *
+ *   @copyright 2016-2018 Huan LI <zixia@zixia.net>
+ *
+ *   Licensed under the Apache License, Version 2.0 (the "License");
+ *   you may not use this file except in compliance with the License.
+ *   You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   Unless required by applicable law or agreed to in writing, software
+ *   distributed under the License is distributed on an "AS IS" BASIS,
+ *   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *   See the License for the specific language governing permissions and
+ *   limitations under the License.
+ *
+ */
+import { test, sinon } from 'tstest'
+
+import { log } from '../src/config.js'
+
+import { PuppetWeChat } from '../src/puppet-wechat.js'
+import { WebMessageMediaPayload, WebMessageType } from '../src/web-schemas.js'
+import { FileBox } from 'file-box'
+import request from 'request'
+import { extname } from 'path'
+
+class PuppetTest extends PuppetWeChat {}
+
+test('Send Attachment', async (t) => {
+  const puppet = new PuppetTest()
+
+  const sandbox = sinon.createSandbox()
+  sandbox.stub(puppet.bridge, 'getCheckUploadUrl').returns(Promise.resolve('getCheckUploadUrl'))
+  sandbox.stub(puppet.bridge, 'getUploadMediaUrl').returns(Promise.resolve('getUploadMediaUrl'))
+  sandbox.stub(puppet.bridge, 'getBaseRequest').returns(Promise.resolve('{}'))
+  sandbox.stub(puppet.bridge, 'getPassticket').returns(Promise.resolve('getPassticket'))
+  sandbox.stub(puppet.bridge, 'cookies').returns(Promise.resolve([]))
+  sandbox.stub(puppet.bridge, 'hostname').returns(Promise.resolve('hostname'))
+  sandbox.replaceGetter(puppet, 'currentUserId', () => 'currentUserId')
+  const conversationId = 'filehelper'
+  const uploadMediaUrl = await puppet.bridge.getUploadMediaUrl()
+  const checkUploadUrl = await puppet.bridge.getCheckUploadUrl()
+  const mockedResCheckUpload = {
+    AESKey: 'AESKey',
+    Signature: 'Signature',
+  }
+  const mockedResUploadMedia = {
+    MediaId: 'MediaId',
+  }
+  const getExtName = (filename: string) => {
+    return extname(filename).slice(1)
+  }
+  const extToType = (ext: string): WebMessageType => {
```

Comment:
Why we re-define this function, instead of using the `puppet.extToType()` for testing?

I think we should test those methods on the `puppet`, please remove the defines in the tests, and use the existing code in puppet instead.

This PR will ready to be merged after we finished improving testing codes.